ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/PartStackPresenter.java

```diff
@@ -496,6 +476,15 @@ public void onTabClicked(@NotNull TabItem selectedTab) {
         activePart = parts.get(selectedTab);
         activePart.onOpen();
         selectActiveTab(activeTab);
+
+        final double workbenchPartSize = workBenchPartController.getSize();
+        if (state != State.MAXIMIZED && isSizeOverridden(workbenchPartSize)) {
+            currentSize = workbenchPartSize;
+        }
+
+        if (activePart != null) {
+            updateWorkBenchPartSize();
+        }
```

Comment:
Could you please add comments on new blocks you are added?
It will help developers to understand the logic of the code.